plugins/device/transaction.go

```diff
@@ -59,3 +61,54 @@ func txnPrepareDevice(c transaction.TxnCtx) error {
 	}
 	return nil
 }
+
+func txnDeleteDevice(c transaction.TxnCtx) error {
+	var peerID string
+	if err := c.Get("peerid", &peerID); err != nil {
+		return err
+	}
+
+	var deviceName string
+	if err := c.Get("device", &deviceName); err != nil {
+		return err
+	}
+
+	devices, err := deviceutils.GetDevices(peerID)
+	if err != nil {
+		return err
+	}
+
+	if len(devices) == 0 {
+		return errors.New("No devices added in the given peer")
+	}
+
+	vgName := ""
+	for _, device := range devices {
+		if device.Device == deviceName {
+			vgName = device.VgName()
+		}
+	}
+
+	if vgName == "" {
+		return errors.New("No device found with given device name")
```

Comment:
an error message should start with the small case. and move this error message to error pkg. it will be helpful for the consumer.